`rptest`: add `nessie` end to end `datalake` testing

WillemKauf opened the pull request:
Now that https://github.com/redpanda-data/redpanda/pull/25230 is merged, we can support end to end `nessie` testing in our `datalake` testing.

Makes a minor change to `iceberg` in the core and a minor change to `datalake_services.py`, and then adds `nessie` as a supported catalog type in order to parameterize all of our end to end `datalake` testing with it.

## Backports Required

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

* none

WillemKauf committed:
`iceberg`: default `stage_commit` in `create_table_request`

Some catalogs, namely `nessie`, don't like it when this attribute is left
unset:

```
Caused by: com.fasterxml.jackson.databind.exc.ValueInstantiationException:
Cannot construct instance of
`org.projectnessie.catalog.formats.iceberg.rest.ImmutableIcebergCreateTableRequest`,
problem: Cannot build IcebergCreateTableRequest,
some of required attributes are not set [stageCreate]
```

WillemKauf committed:
`rptest`: move raising of `NotImplementedError` into `else`

This would previously pass and return `None` if a `catalog_type` was
present in `SUPPORTED_CATALOG_TYPES`, but didn't have a constructor in
an `elif` statement here.

WillemKauf committed:
`rptest`: set `iceberg_disable_snapshot_tagging` for `nessie`

Nessie does not support Iceberg tags. We must set this cluster
config property in order to avoid issues using the catalog.

WillemKauf committed:
`rptest`: use `vendor_api_url` for `nessie` in query engine

Query engines work with the Nessie vendor api endpoint, not its Iceberg
endpoint.

WillemKauf committed:
`rptest`: add `nessie` to `SUPPORTED_CATALOG_TYPES`

And add a constructor in `make_catalog_service_for_type()`

andrwng commented on tests/rptest/tests/datalake/datalake_services.py:
```diff
@@ -83,12 +83,13 @@ def setUp(self):
-                default_warehouse_dir=self.catalog_service.
-                cloud_storage_warehouse,
-                catalog_type=self.catalog_service.catalog_type())
+            catalog_uri = self.catalog_service.vendor_api_url if self.catalog_service.catalog_type(
```
Not suggesting a change yet, since I guess we'll see how other catalogs shake out, but I wonder if it makes sense to assume the vendor API url matches the Iceberg REST URL, and to rebrand the "vendor API url" as the "query engine url" or somesuch. Though I imagine maybe we'd also allow test authors to determine which endpoint they want to use when configuring the query engine.

WillemKauf replied:
Yeah I also don't like the fact this line of code is here, but until we start seeing more catalogs added, I also don't think its worth making an immediate change.

andrwng reviewed (dismissed):
🔥 🔥 🔥

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/62557-->
<!--dee774e5-f53f-4cbf-9bfd-02935c970c4e-->
<!--title-->
## Retry command for [Build#62557](https://buildkite.com/redpanda/redpanda/builds/62557)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/datalake/datalake_e2e_test.py::DatalakeE2ETests.test_remove_expired_snapshots@{"catalog_type":"nessie","cloud_storage_type":1}
```
<!--/body-->

WillemKauf committed:
`rptest`: ignore `nessie` for `test_remove_expired_snapshots`

This test will fail with `nessie` as the catalog type, since it
doesn't support the use of tags.

Add `nessie` parameterizations to the `@ignore` decorator.

WillemKauf commented:
Push to:
* Add `nessie` parameterizations to the `@ignore` decorator for `test_remove_expired_snapshots`, since this test will always fail with `nessie` as the catalog.

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--5a221f0f-c107-49c7-be74-d53daee8c6b1-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62564">build#62564</a></summary>


| test_id                                                                                                                     | test_kind   | job_url                                                                                   | test_status   | passed   |
|:----------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62564#01955e5b-96cb-4674-a67a-2b47f11c73b3 | FLAKY         | 1/3      |
| rptest.tests.partition_balancer_test.PartitionBalancerTest.test_unavailable_nodes                                           | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62564#01955e5e-690c-4b55-a485-7ece3ec73873 | FLAKY         | 1/2      |
</details>
<!--/body-->

nvartolomei commented on tests/rptest/tests/datalake/datalake_e2e_test.py:
```diff
@@ -159,9 +159,16 @@ def test_upload_after_external_update(self, cloud_storage_type,
             catalog_type=supported_catalog_types())
     def test_remove_expired_snapshots(self, cloud_storage_type, catalog_type):
+        """
+        Nessie doesn't support tags, so it is ignored for this test.
```
nit: This comment would be more contextual on the @ignore decorator. Also, it gets printed as test description now in ducktape tests results where it doesn't bear any meaning.
